server/lib/genome/importers/file_importers/nci.rb

```diff
@@ -1,125 +1,55 @@
 module Genome; module Importers; module FileImporters; module Nci;
   class Importer < Genome::Importers::Base
     attr_reader :file_path
-    attr_reader :source_db_name
-    attr_reader :source
-    attr_reader :drug_filter
+    attr_reader :worksheet
 
     def initialize(file_path)
-      @file_path = handle_file_location file_path
-      @source_db_name = 'NCI'
+      @file_path = file_path
     end
 
-    def default_filetype
-      'xml'
+    def get_version
+      source_db_version = Date.today.strftime('%d-%B-%Y')
+      @new_version = source_db_version
     end
 
-    def run_parser
-      @drug_filter = DrugFilter.new
-      parser = Nokogiri::XML::SAX::Parser.new(drug_filter)
-      parser.parse(File.open(@file_path))
+    def import
+      remove_existing_source
+      create_new_source
+      create_interaction_claims
     end
 
-    def create_claims
-      run_parser
+    private
 
-      @drug_filter.all_records.each do |record|
-        gene_claim = create_gene_claim(record[0],'CGI Gene Name')
-        drug_claim = create_drug_claim(record[1],record[1],'NCI Drug Name')
-        create_drug_claim_alias(drug_claim, record[2], 'NCI Drug Code')
-        interaction_claim = create_interaction_claim(gene_claim, drug_claim)
-        create_interaction_claim_publication(interaction_claim, record[3])
-        create_interaction_claim_link(interaction_claim, 'The Cancer Gene Index Gene-Disease and Gene-Compound XML Documents', 'https://wiki.nci.nih.gov/display/cageneindex/The+Cancer+Gene+Index+Gene-Disease+and+Gene-Compound+XML+Documents' )
-        end
-      backfill_publication_information
+    def remove_existing_source
+      Utils::Database.delete_source('NCI')
     end
 
     def create_new_source
       @source ||= Source.create(
-        {
-          base_url: 'https://wiki.nci.nih.gov/display/cageneindex/The+Cancer+Gene+Index+Gene-Disease+and+Gene-Compound+XML+Documents',
-          site_url: 'https://wiki.nci.nih.gov/display/cageneindex',
-          citation: 'https://wiki.nci.nih.gov/display/cageneindex',
-          source_db_version: set_current_date_version,
-          source_db_name: source_db_name,
-          full_name: 'NCI Cancer Gene Index',
-          license: 'Public domain',
-          license_link: 'https://www.cancer.gov/policies/copyright-reuse'
-        }
+          {
+              base_url: 'https://wiki.nci.nih.gov/display/cageneindex/The+Cancer+Gene+Index+Gene-Disease+and+Gene-Compound+XML+Documents',
+              site_url: 'https://wiki.nci.nih.gov/display/cageneindex',
+              citation: 'https://wiki.nci.nih.gov/display/cageneindex',
+              source_db_version:  get_version,
```

Comment:
`get version` should be deleted and we should be hardcoding the data date. Last update on dgidb is listed as "14-September-2017" so we can probably use that